### PR TITLE
HIVE-24433 - AutoCompaction is not getting triggered for CamelCase Partition Values

### PR DESCRIPTION
**Pull request by nareshpr**

### CODE REVIEW

**Comment by pvargacl** [comment]
Hi @nareshpr  thanks for picking this up. Could you add a test case to TestInitiator that covers this use case.
cc @klcopp  @deniskuzZ

**Comment by nareshpr** [comment]
Thanks for looking into this @pvargacl. I added testcase in TestInitiator, please review

File: standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/txn/TxnHandler.java [hunk 1]
```diff
@@ -2725,7 +2725,7 @@ private void insertTxnComponents(long txnid, LockRequest rqst, Connection dbConn
           }
           String dbName = normalizeCase(lc.getDbname());
           String tblName = normalizeCase(lc.getTablename());
-          String partName = normalizeCase(lc.getPartitionname());
+          String partName = lc.getPartitionname();
```
Comment (deniskuzZ) [inline standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/txn/TxnHandler.java:2728]: @nareshpr , do you know if partition name part (<name>=value) is already normalized here?

Comment (nareshpr) [reply standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/txn/TxnHandler.java:2728]: Yes, i validated below 4 SQL's with my patch, partition(key=value) is already normalized 

insert into table abc PARTITION(CitY='Bangalore') values('Dan');
insert overwrite table abc partition(CiTy='Bangalore') select Name from abc;
update table abc set Name='xy' where CiTy='Bangalore';
delete from abc where CiTy='Bangalore';

Comment (deniskuzZ) [reply standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/txn/TxnHandler.java:2728]: Could there be an issue if HMS API is used outside of Hive, for example via Impala?

Comment (nareshpr) [reply standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/txn/TxnHandler.java:2728]: I doubt HMS API's will hit this flow. Let me know any example or testcase. If required, i can change code to split key=value & convert key to lowerCase and join. Let me know if that is required.

Comment (deniskuzZ) [reply standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/txn/TxnHandler.java:2728]: Impala uses HMS API for transaction management. They rely on a TxnHandler.lock() method to handle concurrent txns.

Comment (nareshpr) [reply standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/txn/TxnHandler.java]: I changed it to split & convert partition key to lower key.

File: standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/txn/TxnHandler.java [hunk 2]
```diff
@@ -2877,6 +2877,20 @@ private static String normalizeCase(String s) {
     return s == null ? null : s.toLowerCase();
   }
 
+  private static String normalizePartitionCase(String s) {
+    if (s == null) {
+      return null;
+    } else {
```
Comment (klcopp) [inline standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/txn/TxnHandler.java:2883]: No need for the else clause

**Review by klcopp — COMMENTED** [review]
LGTM, except for one nit

File: standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/txn/TxnHandler.java [hunk 3]
```diff
@@ -2877,6 +2877,20 @@ private static String normalizeCase(String s) {
     return s == null ? null : s.toLowerCase();
   }
 
+  private static String normalizePartitionCase(String s) {
```
Comment (deniskuzZ) [inline standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/txn/TxnHandler.java:2880]: @nareshpr, LGTM, however could you please try to reuse FileUtils.makePartName(List<String> partCols, List<String> vals)

**Review by deniskuzZ — APPROVED** [review]

**Pull request merged**